### PR TITLE
Rasdaman installer

### PR DESCRIPTION
**Pull request by kalxas**
refactored install rasdaman to use the rasdaman installer with an osgeo profile

### CODE REVIEW

**Comment by kalxas** [comment]
Replaces #73

**Comment by kalxas** [comment]
Thanks @woland84 for the actual PR

**Comment by kalxas** [comment]
Unfortunately the new installer did not work:

===============================================================
Starting "install_rasdaman.sh" ...
===============================================================
--2016-02-12 22:56:48--  http://download.rasdaman.org/installer/install.sh
Resolving download.rasdaman.org (download.rasdaman.org)... 212.201.49.173
Connecting to download.rasdaman.org (download.rasdaman.org)|212.201.49.173|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8871 (8.7K) [text/x-sh]
Saving to: 'install.sh'

     0K ........                                              100% 7.66M=0.001s

2016-02-12 22:56:48 (7.66 MB/s) - 'install.sh' saved [8871/8871]

[0mChecking if installer needs to update...[0m
[0mInstaller is at latest version.[0m
[0mStrategy building is handled by the General Strategy Builder.[0m
[1mReview the installation settings:[0m
[0m
Installation path: /opt/rasdaman/
Version: master
Update: False
User: user
Debug: False
Communication protocol: rasnet
Database type: sqlite
  Database data directory: /opt/rasdaman/data/
Rasdaman binary: http://download.rasdaman.org/packages/binary/rasdaman.tar.gz
Petascope enabled: True
  Petascope database url: jdbc:postgresql://localhost:5432/petascopedb
  Petascope database user: rasdaman
  Petascope server type: tomcat
  Tomcat home: /var/lib/tomcat7/
Secore enabled: True
Run systemtest: False
Rasdaman demo data: True
Petascope demo data: True
[0m
[1mValidating system...[0m
[0mCleaning up any old rasdaman files...[0m
[31mAn error has occured while executing this command: dropdb petascopedb. We collected the following information:
Return code: 1
Stderr: dropdb: database removal failed: ERROR:  database "petascopedb" does not exist

Stdout:[0m
ERROR: The installer returned an error code. The installation might not be valid.

**Comment by woland84** [comment]
Fixed, could you try running it again?

**Comment by kalxas** [comment]
Starting over

**Comment by kalxas** [comment]
===============================================================
Starting "install_rasdaman.sh" ...
===============================================================
--2016-02-16 12:59:03--  http://download.rasdaman.org/installer/install.sh
Resolving download.rasdaman.org (download.rasdaman.org)... 212.201.49.173
Connecting to download.rasdaman.org (download.rasdaman.org)|212.201.49.173|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8871 (8.7K) [text/x-sh]
Saving to: 'install.sh'

     0K ........                                              100% 9.69M=0.001s

2016-02-16 12:59:04 (9.69 MB/s) - 'install.sh' saved [8871/8871]

[0mChecking if installer needs to update...[0m
[0mInstaller is at latest version.[0m
[0mStrategy building is handled by the General Strategy Builder.[0m
[1mReview the installation settings:[0m
[0m
Installation path: /opt/rasdaman/
Version: master
Update: False
User: user
Debug: False
Communication protocol: rasnet
Database type: sqlite
  Database data directory: /opt/rasdaman/data/
Rasdaman binary: http://download.rasdaman.org/packages/binary/rasdaman.tar.gz
Petascope enabled: True
  Petascope database url: jdbc:postgresql://localhost:5432/petascopedb
  Petascope database user: rasdaman
  Petascope server type: tomcat
  Tomcat home: /var/lib/tomcat7/
Secore enabled: True
Run systemtest: False
Rasdaman demo data: True
Petascope demo data: True
[0m
[1mValidating system...[0m
[0mCleaning up any old rasdaman files...[0m
[0mOld rasdaman files cleaned up successfully.[0m
[1mSystem was validated.[0m
[1mInstalling third party dependencies...[0m
[0mChecking user user....[0m
[0mUser user checked successfully.[0m
[0mCreating directory structure...[0m
[0mDirectory structure created successfully.[0m
[0mInstalling deb packages ['make', 'zlib1g', 'zlib1g-dev', 'automake', 'autotools-dev', 'libtool', 'm4', 'gawk', 'bison', 'flex', 'git-core', 'g++', 'libncurses5-dev', 'libtiff-dev', 'libgdal-dev', 'gdal-bin', 'libnetpbm10-dev', 'libjpeg-dev', 'libpng-dev', 'libffi-dev', 'postgresql', 'postgresql-contrib', 'libedit-dev', 'libecpg-dev', 'libsqlite-dev', 'sqlite3', 'python-dateutil', 'python-lxml', 'python-pip', 'python-gdal', 'libboost-all-dev', 'openjdk-7-jdk', 'python-pip', 'unzip', 'tomcat7']... [0m
[0mDeb packages installed successfully[0m
[0mInstalling pip packages ['glob2']... [0m
[0mPip packages installed successfully[0m
[0mSetting up the PATH environment variable...[0m
[0mThe PATH environment variable was set up successfully.[0m
[0mAdding the current user to tomcat user group...[0m
[0mUser user added to tomcat user group successfully.[0m
[0mIncreasing tomcat memory limit...[0m
[0mTomcat memory increased succesfully.[0m
[1mThird party dependencies installed successfully.[0m
[1mRetrieving rasdaman...[0m
[1mRasdaman retrieved successfully[0m
[1mBuilding rasdaman...[0m
[0mDownloading and installing rasdaman binaries...[0m
[0mDownloaded and installed rasdaman binaries successfully[0m
[0mDownloading and deploying petascope war...[0m
[0mDownloaded and installed petascope war...[0m
[0mDownloading and deploying secore war...[0m
[0mDownloaded and installed secore war...[0m
[1mRasdaman was built successfully.[0m
[1mPreparing rasdaman...[0m
[0mPreparing Rasdaman SQLite database...[0m
[0mRasdaman SQLite database was prepared.[0m
[0mPreparing petascope database...[0m
[0mPetascope database was prepared.[0m
[1mRasdaman prepared successfully.[0m
[1mFinalizing installation...[0m
[0mCustomizing rasdaman for OSGeo Live...[0m
[0mConfiguring tomcat to autodeploy war...[0m
[0mTomcat configured successfully.[0m
[0mCreating starting scripts...[0m
[0mStarting scripts successfully created.[0m
[0mCreating desktop icons...[0m
[0mRasdaman demo data installed successfully.[0m
[0mPetascope demo data installed successfully.[0m
[31mAn unexpected error has occured. Please provide the below information to a developer:[0m
[31m[Errno 2] No such file or directory: '/home/user/Desktop/Databases/start_rasdaman_server.desktop'[0m
Traceback (most recent call last):
  File "/tmp//rasdaman-installer/main.py", line 93, in main
    test_result = strategy.run()
  File "/tmp/rasdaman-installer/strategies/base_strategy.py", line 82, in run
    self.finalize()
  File "/tmp/rasdaman-installer/strategies/base_strategy.py", line 129, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 304, in finalize
    FinalizerDecorator.finalize(self)
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 77, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 327, in finalize
    FinalizerDecorator.finalize(self)
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 77, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 210, in finalize
    FinalizerDecorator.finalize(self)
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 77, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 116, in finalize
    FinalizerDecorator.finalize(self)
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 77, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 92, in finalize
    FinalizerDecorator.finalize(self)
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 77, in finalize
    self.finalizer.finalize()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 523, in finalize
    self.create_desktop_applications()
  File "/tmp/rasdaman-installer/finalizers/finalizers.py", line 427, in create_desktop_applications
    fp = open(complete_path, "w")
IOError: [Errno 2] No such file or directory: '/home/user/Desktop/Databases/start_rasdaman_server.desktop'
ERROR: The installer returned an error code. The installation might not be valid.

**Comment by woland84** [comment]
Hey,

In the previous osgeo there was a folder on the Desktop called databases? Does it not exist anymore in this version? Or should we created it if it does not exist?

Thanks for the help!

**Comment by kalxas** [comment]
@woland84 that gets created after the rasdaman installation.
All menus and system settings are set at the end of the build.

**Comment by kalxas** [comment]
@woland84 I see that this new installer hides lot of the installation details and it won't help much with debugging. I need to make a deep review of what happens under the hood...

**Comment by kalxas** [comment]
why not use the packaged version?
http://rasdaman.org/wiki/InstallFromDEB

**Comment by kalxas** [comment]
ok, I see that the installer is the new default way to install rasdaman (including profiles)
http://rasdaman.org/wiki/Installer

**Comment by woland84** [comment]
In the previous script the rasdaman desktop entries were created in the following two paths:["/usr/local/share/applications/", "/home/user/Desktop/Databases/"]. Should we only create in /usr now?

You can review the installer by running 
```bash
cd /tmp
wget http://download.rasdaman.org/installer/rasdaman-installer.zip
unzip rasdaman-installer.zip
# to run the osgeo profile
cd rasdaman-installer
sudo python main.py -p profiles/installer/osgeo.json
```

It might be a bit unorthodox but it helps us keep the build process under one central method. The installer can install by itself, but also generates the packages that you see on the rasdaman download pages and is even used in our Jenkins machine to build, install and test rasdaman. This setup allows us to have one well tested codebase for all the installations that we do (be them via the installer or via a debian package) with all the possible options that we have (e.g. rasdaman stand-alone as an array-database, rasdaman + petascope as a geo-database, rasdaman with a postgres vs sqlite backend and so on).

The debian package works nicely but it doesn't have the customizations needed for osgeo (e.g. the rasdaman-demo website, the custom datasets that we integrate, desktop files etc). We could create a package specifically for osgeo, but it would be basically the installer in a .deb format :)

Hope it makes some sense and sorry for wasting your time with all this stuff!

Thanks,
Alex

**Comment by kalxas** [comment]
Hi Alex,
Thanks for the clarifications.

Some comments regarding desktop files:
In the previous script, the desktop files where created in "/usr/local/share/applications/" and then copied to "/home/user/Desktop"
https://github.com/OSGeo/OSGeoLive/blob/master/bin/install_rasdaman.sh#L333
https://github.com/OSGeo/OSGeoLive/blob/master/bin/install_rasdaman.sh#L347
https://github.com/OSGeo/OSGeoLive/blob/master/bin/install_rasdaman.sh#L361

The copy happens here:
https://github.com/OSGeo/OSGeoLive/blob/master/bin/install_rasdaman.sh#L374

Then, after all installers finish, there is a script doing the cleanup of launchers:
https://github.com/OSGeo/OSGeoLive/blob/master/bin/install_icons_and_menus.sh#L64

We need to keep this working the same way.
Let the installer only create those files in /usr and I will make the rest happen.

The original script also had some other parameters, e.g. how can we control the tomcat reboot time? Tomcat is very sensitive in our setup, since it hosts many other war files and takes almost 60 secs to restart on the build server.

I understand the need to streamline the installation process.
Ideally I would prefer a deb package in a ppa (including source files) to copy from and customize/hack in order to include our post-install configuration.
If that is too much work, we can use the installer...

**Comment by kalxas** [comment]
Looking at the OSGeo profile in the installer I see this:

{
  "user": "user",
  "version": "master",
  "osgeo": true,
  "automated": true,
  "database": {
    "type": "sqlite"
  },
  "build": {
    "type": "binary",
    "rasdaman_binary_url": "http://download.rasdaman.org/packages/binary/rasdaman.tar.gz",
    "petascope_binary_url": "http://download.rasdaman.org/packages/binary/rasdaman.war",
    "secore_binary_url": "http://download.rasdaman.org/packages/binary/def.war"
  },
  "communication_protocol": "rasnet",
  "install_path": "/opt/rasdaman",
  "debug": false,
  "petascope": {
    "enabled": true,
    "type": "tomcat",
    "database_conf": {
      "url": "jdbc:postgresql://localhost:5432/petascopedb",
      "user": "rasdaman"
    }
  },
  "secore": {
    "enabled": true
  },
  "demo": {
    "rasdaman_data": true,
    "petascope_data": true
  }
}

Does this pull a tagged branch from git or "master"?

**Comment by kalxas** [comment]
oh, this profile is not building from source :)
please ignore my last question

**Comment by woland84** [comment]
I updated the profile to install the desktop entries in "/usr/local/share/applications/". Running the install script again should fix the previous issue.

I will investigate the ppa option, we tried it before but it didn't want to properly compile grpc, now i see there is a grpc package so it might be an option though I don't think it will be ready in time. Worst case I can adapt the previous install script to work with rasdaman 9.2 (least preferable option in my opinion).

About tomcat, we restart it twice during the installation, but we wait for it to properly restart before continuing so I would doubt that it would cause an issue.

Let me know what you think.

**Comment by kalxas** [comment]
Thanks Alex, 
I will try again and let you know the result.

**Comment by kalxas** [comment]
===============================================================
Starting "install_rasdaman.sh" ...
===============================================================
--2016-02-16 21:42:48--  http://download.rasdaman.org/installer/install.sh
Resolving download.rasdaman.org (download.rasdaman.org)... 212.201.49.173
Connecting to download.rasdaman.org (download.rasdaman.org)|212.201.49.173|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8871 (8.7K) [text/x-sh]
Saving to: 'install.sh'

     0K ........                                              100% 5.25M=0.002s

2016-02-16 21:42:48 (5.25 MB/s) - 'install.sh' saved [8871/8871]

Checking if installer needs to update...
Installer is at latest version.
Strategy building is handled by the General Strategy Builder.
Review the installation settings:

Installation path: /opt/rasdaman/
Version: master
Update: False
User: user
Debug: False
Communication protocol: rasnet
Database type: sqlite
  Database data directory: /opt/rasdaman/data/
Rasdaman binary: http://download.rasdaman.org/packages/binary/rasdaman.tar.gz
Petascope enabled: True
  Petascope database url: jdbc:postgresql://localhost:5432/petascopedb
  Petascope database user: rasdaman
  Petascope server type: tomcat
  Tomcat home: /var/lib/tomcat7/
Secore enabled: True
Run systemtest: False
Rasdaman demo data: True
Petascope demo data: True

Validating system...
Cleaning up any old rasdaman files...
Old rasdaman files cleaned up successfully.
System was validated.
Installing third party dependencies...
Checking user user....
User user checked successfully.
Creating directory structure...
Directory structure created successfully.
Installing deb packages ['make', 'zlib1g', 'zlib1g-dev', 'automake', 'autotools-dev', 'libtool', 'm4', 'gawk', 'bison', 'flex', 'git-core', 'g++', 'libncurses5-dev', 'libtiff-dev', 'libgdal-dev', 'gdal-bin', 'libnetpbm10-dev', 'libjpeg-dev', 'libpng-dev', 'libffi-dev', 'postgresql', 'postgresql-contrib', 'libedit-dev', 'libecpg-dev', 'libsqlite-dev', 'sqlite3', 'python-dateutil', 'python-lxml', 'python-pip', 'python-gdal', 'libboost-all-dev', 'openjdk-7-jdk', 'python-pip', 'unzip', 'tomcat7']... 
Deb packages installed successfully
Installing pip packages ['glob2']... 
Pip packages installed successfully
Setting up the PATH environment variable...
The PATH environment variable was set up successfully.
Adding the current user to tomcat user group...
User user added to tomcat user group successfully.
Increasing tomcat memory limit...
Tomcat memory increased succesfully.
Third party dependencies installed successfully.
Retrieving rasdaman...
Rasdaman retrieved successfully
Building rasdaman...
Downloading and installing rasdaman binaries...
Downloaded and installed rasdaman binaries successfully
Downloading and deploying petascope war...
Downloaded and installed petascope war...
Downloading and deploying secore war...
Downloaded and installed secore war...
Rasdaman was built successfully.
Preparing rasdaman...
Preparing Rasdaman SQLite database...
Rasdaman SQLite database was prepared.
Preparing petascope database...
Petascope database was prepared.
Rasdaman prepared successfully.
Finalizing installation...
Customizing rasdaman for OSGeo Live...
Configuring tomcat to autodeploy war...
Tomcat configured successfully.
Creating starting scripts...
Starting scripts successfully created.
Creating desktop icons...
Desktop icons created successfully.
Striping binaries and removing unneeded files for OSGeo Live...
Binaries stripped successfully.
Deploying local earthlook...
Local earthlook deployed successfully.
Ingesting demo data...
Rasdaman demo data installed successfully.
Petascope demo data installed successfully.
An error has occured while executing this command: /opt/rasdaman//bin/wcst_import.sh /tmp/ingest/NIR. We collected the following information:                                       
Return code: 1                                                                            
Stderr:                                                                                   
Stdout:Initialization
Collected files: []...

Validation
An error has occured in the execution of the program. Error Message: [Errno socket error] [Errno 111] Connection refused                                                            
Stack Trace: Traceback (most recent call last):                                           
  File "/opt/rasdaman/share/rasdaman/wcst_import/wcst_import.py", line 126, in main       
    reg.run_recipe(session)                                                               
  File "/opt/rasdaman/share/rasdaman/wcs                                                  
ERROR: The installer returned an error code. The installation might not be valid.

**Comment by woland84** [comment]
We managed to synchronize our data server restart with your build session. The server is again up and running :)

**Comment by kalxas** [comment]
ok thanks, will try again :)

**Comment by kalxas** [comment]
This is the latest output:

===============================================================
Starting "install_rasdaman.sh" ...
===============================================================
--2016-02-17 01:04:25--  http://download.rasdaman.org/installer/install.sh
Resolving download.rasdaman.org (download.rasdaman.org)... 212.201.49.173
Connecting to download.rasdaman.org (download.rasdaman.org)|212.201.49.173|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8871 (8.7K) [text/x-sh]
Saving to: 'install.sh'

     0K ........                                              100% 8.86M=0.001s

2016-02-17 01:04:26 (8.86 MB/s) - 'install.sh' saved [8871/8871]

[0mChecking if installer needs to update...[0m
[0mInstaller is at latest version.[0m
[0mStrategy building is handled by the General Strategy Builder.[0m
[1mReview the installation settings:[0m
[0m
Installation path: /opt/rasdaman/
Version: master
Update: False
User: user
Debug: False
Communication protocol: rasnet
Database type: sqlite
  Database data directory: /opt/rasdaman/data/
Rasdaman binary: http://download.rasdaman.org/packages/binary/rasdaman.tar.gz
Petascope enabled: True
  Petascope database url: jdbc:postgresql://localhost:5432/petascopedb
  Petascope database user: rasdaman
  Petascope server type: tomcat
  Tomcat home: /var/lib/tomcat7/
Secore enabled: True
Run systemtest: False
Rasdaman demo data: True
Petascope demo data: True
[0m
[1mValidating system...[0m
[0mCleaning up any old rasdaman files...[0m
[0mOld rasdaman files cleaned up successfully.[0m
[1mSystem was validated.[0m
[1mInstalling third party dependencies...[0m
[0mChecking user user....[0m
[0mUser user checked successfully.[0m
[0mCreating directory structure...[0m
[0mDirectory structure created successfully.[0m
[0mInstalling deb packages ['make', 'zlib1g', 'zlib1g-dev', 'automake', 'autotools-dev', 'libtool', 'm4', 'gawk', 'bison', 'flex', 'git-core', 'g++', 'libncurses5-dev', 'libtiff-dev', 'libgdal-dev', 'gdal-bin', 'libnetpbm10-dev', 'libjpeg-dev', 'libpng-dev', 'libffi-dev', 'postgresql', 'postgresql-contrib', 'libedit-dev', 'libecpg-dev', 'libsqlite-dev', 'sqlite3', 'python-dateutil', 'python-lxml', 'python-pip', 'python-gdal', 'libboost-all-dev', 'openjdk-7-jdk', 'python-pip', 'unzip', 'tomcat7']... [0m
[0mDeb packages installed successfully[0m
[0mInstalling pip packages ['glob2']... [0m
[0mPip packages installed successfully[0m
[0mSetting up the PATH environment variable...[0m
[0mThe PATH environment variable was set up successfully.[0m
[0mAdding the current user to tomcat user group...[0m
[0mUser user added to tomcat user group successfully.[0m
[0mIncreasing tomcat memory limit...[0m
[0mTomcat memory increased succesfully.[0m
[1mThird party dependencies installed successfully.[0m
[1mRetrieving rasdaman...[0m
[1mRasdaman retrieved successfully[0m
[1mBuilding rasdaman...[0m
[0mDownloading and installing rasdaman binaries...[0m
[0mDownloaded and installed rasdaman binaries successfully[0m
[0mDownloading and deploying petascope war...[0m
[0mDownloaded and installed petascope war...[0m
[0mDownloading and deploying secore war...[0m
[0mDownloaded and installed secore war...[0m
[1mRasdaman was built successfully.[0m
[1mPreparing rasdaman...[0m
[0mPreparing Rasdaman SQLite database...[0m
[0mRasdaman SQLite database was prepared.[0m
[0mPreparing petascope database...[0m
[0mPetascope database was prepared.[0m
[1mRasdaman prepared successfully.[0m
[1mFinalizing installation...[0m
[0mCustomizing rasdaman for OSGeo Live...[0m
[0mConfiguring tomcat to autodeploy war...[0m
[0mTomcat configured successfully.[0m
[0mCreating starting scripts...[0m
[0mStarting scripts successfully created.[0m
[0mCreating desktop icons...[0m
[0mDesktop icons created successfully.[0m
[0mStriping binaries and removing unneeded files for OSGeo Live...[0m
[0mBinaries stripped successfully.[0m
[0mDeploying local earthlook...[0m
[0mLocal earthlook deployed successfully.[0m
[0mIngesting demo data...[0m
[0mRasdaman demo data installed successfully.[0m
[0mPetascope demo data installed successfully.[0m
[31mAn error has occured while executing this command: /opt/rasdaman//bin/wcst_import.sh /tmp/ingest/NIR. We collected the following information:
Return code: 1
Stderr: 
Stdout:[1mInitialization[0m
[0mCollected files: []...[0m
[1m
Validation[0m
[0mThe recipe has been validated and is ready to run.[0m
[0m[1mRecipe:[0m wcs_extract[0m
[0m[1mCoverage:[0m NIR[0m
[0m[1mWCS Service:[0m http://localhost:8080/rasdaman/ows[0m
[0m[1mOperation:[0m INSERT[0m
[0m[1mSubset Correction:[0m False[0m
[0m[1mMocked:[0m False[0m
[0m[1mTrack files:[0m True[0[0m
ERROR: The installer returned an error code. The installation might not be valid.

**Comment by kalxas** [comment]
I see some very concerning result: after enabling the new installer, the iso is broken: autologin does not work and in the login screen, a "rasdaman" account is the default.
Does the new installer create a rasdaman account?

**Comment by woland84** [comment]
Could you send me the /tmp/rasdaman_command_log to check why the data is not importing? 

I just checked the profile and it seems indeed that the petascope component was creating a rasdaman account for postgres (which needs a linux user to use the postgres ident auth method). I will set it as a system user to make sure it does not appear anywhere else.

Thanks,
Alex

**Comment by kalxas** [comment]
Hi Alex,

Unfortunately, the build process of OSGeoLive does not work like that.
We use a chain of installer scripts that are executed in a chroot environment, the iso gets packages based on that chroot, while our build logs are redirected outside the chroot (to avoid deletion).
In your case, your installer saves the output in /tmp/rasdaman_command_log within the chroot environment, which is automatically deleted at the end of the build process. So it is not possible for me to provide you this log.
This is exactly the reason why I stated earlier that debugging will be too difficult if the installer is not within the install_rasdaman.sh

Also, when in chroot, we create a "user" account, we install everything, then we delete this account because this is how Ubuntu live session works. 
When the live disk starts up a new "user" account gets created automatically by the system.

Bottom line: creation of a "rasdaman" user is not permitted, else the Ubuntu live session breaks.

**Comment by woland84** [comment]
I don't understand why a rasdaman (system) user would break the system. Tomcat does the same, so does apache with www-data and so on. The error in the installer was that the rasdaman user was not created as a proper system user (i.e. uid < 999 thus not showing in the login box). Is there something else that I am missing?

Thanks for bearing with me :)

**Comment by kalxas** [comment]
Thanks for the fix, I will try again.

**Comment by woland84** [comment]
wait, it's not commited yet. I'll try it myself with the chroot today so that you don't waste time again.

**Comment by kalxas** [comment]
The previous build has not completed yet, so there is time, no worries.

**Comment by woland84** [comment]
How long does it take normally? I just started it on single core 4GB vm, what should i expect in terms of time it takes to build the iso?

**Comment by kalxas** [comment]
It usually takes 2-3h, but the ubuntu repositories are very slow the last 2 days

**Comment by kalxas** [comment]
I am also doing a rebuild just in case

**Comment by woland84** [comment]
Hey, how did that go? My build failed before getting to rasdaman due to some package dependencies, I'm guessing I am doing something wrong?!
I followed the instructions here https://wiki.osgeo.org/wiki/Live_GIS_Build on a Ubuntu Desktop 14.04

**Comment by kalxas** [comment]
You need a Lubuntu VM without updates to build the iso
The build result is here:
http://aiolos.survey.ntua.gr/gisvm/dev/osgeo-live-nightly-build98-amd64-097d5c5-log.tar.gz

**Comment by woland84** [comment]
Ah, the build page should be updated I guess, I assumed it works with vanilla ubuntu as well from this sentence:
> All you need is a running Ubuntu/Xubuntu/Kubuntu/Lubuntu installation (even within a virtual machine as long as it has ~20GB free disk space). All needed to be done are the following steps under a "user" account:

Seems it worked correctly in the build log, any major error you encountered with the overall iso? Also, could I get the resulting iso?

Thank you for all the help!

**Comment by kalxas** [comment]
I am also downloading now to test:
http://aiolos.survey.ntua.gr/gisvm/dev/osgeo-live-nightly-build98-amd64-097d5c5.iso

**Comment by kalxas** [comment]
@woland84 unfortunately the live session is again broken.
See this in log:

===============================================================
Finished "setdown.sh"
Disk Usage1: setdown.sh,Filesystem,1K-blocks,Used,Available,Use%,Mounted_on,date
Disk Usage2: setdown.sh,-,40197,16699,21434,44%,/,2016-02-19 00:32:52+00:00
Temp Usage: setdown.sh,2297	/tmp
===============================================================
Looking for files to backup/remove ...
Removing files ...
Removing user `user' ...
Warning: group `user' has no more members.
userdel: user user is currently used by process 894
/usr/sbin/deluser: `/usr/sbin/userdel user' returned error code 8. Exiting.
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
user:x:1000:1000:user,,,:/home/user:/bin/bash
Removing 'local diversion of /sbin/initctl to /sbin/initctl.distrib'
umount: /proc: device is busy.
        (In some cases useful info about processes that use
         the device is found by lsof(8) or fuser(1))


On a normal diff this is the log part:
===============================================================
Finished "setdown.sh"
Disk Usage1: setdown.sh,Filesystem,1K-blocks,Used,Available,Use%,Mounted_on,date
Disk Usage2: setdown.sh,-,40197,18245,19888,48%,/,2016-02-18 01:00:43+00:00
Temp Usage: setdown.sh,3906	/tmp
===============================================================
Looking for files to backup/remove ...
Removing files ...
Removing user `user' ...
Warning: group `user' has no more members.
Done.
nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
Removing 'local diversion of /sbin/initctl to /sbin/initctl.distrib'
umount: /proc: device is busy.
        (In some cases useful info about processes that use
         the device is found by lsof(8) or fuser(1))

So it seems that some rasdaman process is blocking user account deletion.
Please make sure that everything is down when you finish your installation (tomcat, rasdaman servers, everything), like it was in the past installer.

**Comment by kalxas** [comment]
Hi Alex, @woland84 
Any news on this issue?
Would you need a new build to test?

**Comment by kalxas** [comment]
Still the same problem on my latest build.

**Comment by woland84** [comment]
sorry for the late response, was away the last couple of days, I just wrote a fix and testing right, will let you know after I commit (~ 1hour). Thanks!

**Comment by woland84** [comment]
Ok seems both are closed now, hopefully this is the issue. Could you retry the build?

**Comment by kalxas** [comment]
Thanks Alex, will rebuild now

**Comment by kalxas** [comment]
@woland84 it now works as expected.
Thank you for all your effort.
Merging now

**Comment by woland84** [comment]
Thank you for all your help! Always a pleasure collaborating :)
Let me know when a new build is available so I can do some checks as well.

**Comment by kalxas** [comment]
http://aiolos.survey.ntua.gr/gisvm/dev/osgeo-live-nightly-build127b-amd64-15a9e57.iso

**Comment by kalxas** [comment]
Unfortunately, the next builds seem to have the issue again: user is not deleted and live session is broken.
I will do another build from master and if this persists I will revert this PR

**Comment by kalxas** [comment]
New build that still fails:
http://aiolos.survey.ntua.gr/gisvm/dev/osgeo-live-nightly-build127-pr74-amd64-15a9e57.iso

**Comment by kalxas** [comment]
The commit has been reverted in 2e433c7fb80dd9a695257bc22500b744cd4122ec

**Comment by kalxas** [comment]
For simplicity reasons the iso has been renamed to http://aiolos.survey.ntua.gr/gisvm/dev/osgeo-live-nightly-pr74-amd64-15a9e57.iso
